activerecord/lib/active_record/associations/join_dependency/join_association.rb

```diff
@@ -54,12 +54,18 @@ def join_constraints(foreign_table, foreign_klass, node, join_type, tables, scop
             end
             scope_chain_index += 1
 
-            relation = ActiveRecord::Relation.create(
-              klass,
-              table,
-              predicate_builder,
-            )
-            scope_chain_items.concat [klass.send(:build_default_scope, relation)].compact
+            klass_scope =
+              if klass.current_scope
+                klass.current_scope.clone
+              else
+                relation = ActiveRecord::Relation.create(
+                  klass,
+                  table,
+                  predicate_builder,
+                )
+                klass.send(:build_default_scope, relation)
+              end
```

Comment:
Woops - I'm still figuring out github and just worked on the same issue today.  @sgrif is right in thinking that `klass.all.spawn` will return the relation with default_scope applied if it hasn't been scoped by a block.   The issue is that `klass.all.spawn.table` is not necessarily the Arel::Table you should be using, because it doesn't have the join alias associated with it.  This is what I came up with as a fix: https://github.com/rails/rails/pull/20937